python/venv/private/venv_common.bzl

```diff
@@ -431,25 +432,25 @@ def _create_python_zip_file(
     if not interpreter:
         fail("Unable to locate interpreter from py_toolchain: {}".format(py_toolchain))
 
-    if name == None:
-        name = ctx.label.name
-
     venv_config_info = _create_venv_config_info(
         label = ctx.label,
-        name = name.replace("/", "_"),
+        name = ctx.label.name.replace("/", "_"),
         imports = py_info.imports.to_list(),
     )
 
-    venv_runfiles = ctx.runfiles(transitive_files = depset(transitive = [
-        depset([
-            main,
-            venv_toolchain.process_wrapper,
-        ]),
+    venv_runfiles = depset([
+        main,
+        venv_toolchain.process_wrapper,
+        venv_toolchain.zipapp_main,
+        files_to_run.runfiles_manifest,
+        files_to_run.repo_mapping_manifest,
+    ], transitive = [
         py_runtime.files,
         py_info.transitive_sources,
-    ])).merge(runfiles)
+        runfiles.files,
+    ])
 
-    python_zip_file = ctx.actions.declare_file("{}.pyz".format(name))
+    python_zip_file = ctx.actions.declare_file("{}.pyz".format(ctx.label.name))
```

Comment:
Looking at the consumer of this:
```python
py_venv_zipapp(
    name = "writer_pyz",
    binary = ":writer",
)
```
I would want this to produce a file `writer.pyz` but currently it'll be `writer_pyz.pyz` which is silly and I can't recycle `writer` since it's the name of the `py_venv_binary` now. So maybe this can either conditionally check to ensure the file ends with `.pyz` so users can set it if they want or the rule will ensure the extension is correct?

Reply:
Conditionally added the pyz extension